Assets/Tests/PlayMode/JsonConverterTests.cs

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2024 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+namespace PlayEveryWare.EpicOnlineServices.Tests
+{
+    using Epic.OnlineServices;
+    using Newtonsoft.Json;
+    using NUnit.Framework;
+    using PlayEveryWare.EpicOnlineServices.JsonConverters;
+    using System;
+    using System.Text;
+    using UnityEngine.UI;
+
+    [TestFixture]
+    public class Utf8StringConverterTests
+    {
+        private JsonSerializerSettings _settings;
+
+        [SetUp]
+        public void Setup()
+        {
+            _settings = new JsonSerializerSettings();
+            _settings.Converters.Add(new Utf8StringConverter());
+        }
+
+        [Test]
+        public void Serialize_Utf8String_With_Bytes_Writes_Byte_Array()
+        {
+            // Arrange
+            var utf8String = new Utf8String(new byte[] { 72, 101, 108, 108, 111, 0 });
+
+            // Act
+            var json = JsonConvert.SerializeObject(utf8String, _settings);
+
+            // Assert
+            Assert.AreEqual("[72,101,108,108,111,0]", json);
+        }
+
+        [Test]
+        public void Serialize_Utf8String_With_String_Deserializes_To_Same_String()
+        {
+            string normalString = "blah blah blah";
+            
+            Utf8String test = new Utf8String(normalString);
+
+            var json = JsonConvert.SerializeObject(test, _settings);
+
+            Utf8String deserializedTest = JsonConvert.DeserializeObject<Utf8String>(json, _settings);
+
+            // TODO: I would argue that this is a bug in Epic's code (because this test fails)
```

Comment:
Yup, the test fails. I wonder what should be done? I'm not content with the idea of having a test that fails that *in theory* should succeed but isn't for reasons we can never change... What should be done here?